packages/grid/_modules_/grid/components/cell/GridRowCells.tsx

```diff
@@ -96,7 +96,7 @@ export const GridRowCells = React.memo(function GridRowCells(props: RowCellsProp
     const cellProps: GridCellProps = {
       value: cellParams.value,
       field: column.field,
-      width: column.width!,
+      width: column.minWidth ? Math.max(column.minWidth, column.width!) : column.width!,
```

Comment:
When is this line usefull (same thing for its equivalent in `GridColumnHeaderItem`)
Shouldn't the logic in `updateColumnsWidth` be enough ?
When I remove it, the `Resizing` doc still works.

Reply:
Actually, this is needed for the initial load - the reason is that `updateColumnsWidth` is only called when there is a change. If this is removed then setting `minWidth` won't be respected on the first load.

Reply:
When do we set the flex column width for the 1st load then ?

Reply:
Actually my bad, it is called it's just only respecting `fluid` columns

Reply:
Oh indeed
So I suppose we want to improve the `updateColumnsWidth` (and maybe rename it `preprocessColumnsWidth` or something like that).

Reply:
> Example: minWidth > width, you update the width accordingly in updateColumnsWidth, so width = minWidth and then the original value of the width is lost.

Why is it an issue to lose the original value?

Reply:
I can agree that modifying the column like we do feels wrong.
For instance, we have to put `column.width!` everywhere because we know that the typing is wrong and that after this point the width is always defined. 
I would prefer having an explicit type change.

For instance
```
interface PreProcessedGridColDef extends GridColDef {
 processedValues: {
   width: number
 }
}

const preprocessColumns: (rawColumns: GridColDef[]): PreProcessedGridColDef[]
```

Reply:
The `useGridColumns` already updates the widths in the initial load, but it doesn't anything when there's no fluid column. If we add an else case to the conditional below, we'll be able to compute the widths for non-fluid columns and remove this leaking logic. 

https://github.com/mui-org/material-ui-x/blob/3c0082e6ae06790fde25e4f2772b892162ffaa89/packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts#L55

Reply:
Updated and removed